charts/internal/cilium/charts/config/templates/configmap.yaml

```diff
@@ -336,7 +336,9 @@ data:
 
   enable-ipv4-masquerade: {{ .Values.global.enableIpv4Masquerade | quote }}
   enable-ipv6-masquerade: {{ .Values.global.enableIpv6Masquerade | quote }}
+{{- if not .Values.global.snatToUpstreamDNS.enabled }}
   enable-bpf-masquerade:  {{ .Values.global.enableBPFMasquerade | quote }}
+{{- end }}
```

Comment:
What is the defaulting logic here? Can we end up with having `enable-ipv(4|6)-masquerade` being `false` and `enable-bpf-masquerade` not being set?

Reply:
`enable-ipv4-masquerade` should be always enabled
 `enable-ipv6-masquerade ` is only required for ipv5
 and enable-bpf-masquerade should only be set if we run without overlay but directly depends on snatToUpstreamDNS.

Reply:
No, its globally set to true and we don't override it somewhere.